app/controllers/application_controller.rb

```diff
@@ -5,6 +5,95 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    #checks if logged in
+    if session[:user_id]
```

Comment:
I'd build some application-wide Sinatra [`helpers`](http://www.sinatrarb.com/intro.html#Helpers) like `logged_in?` and `current_user` to abstract some of these ideas out into named methods I can re-use.